modules/aggregation-layers/src/heatmap-layer/triangle-layer.js

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015 - 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import GL from '@luma.gl/constants';
+import {Model, Geometry} from '@luma.gl/core';
+import {Layer} from '@deck.gl/core';
+import vs from './triangle-layer-vertex.glsl';
+import fs from './triangle-layer-fragment.glsl';
+
+const defaultProps = {
+  // data {positions(Buffer), texCoords(Buffer)}
+  count: 0, // number of triangles to be rendered
+  texture: null
+};
+
+export default class TriangleLayer extends Layer {
```

Comment:
Ideally you don't need another layer to render this. You can easily create the model inside `HeatmapLayer` and update its attributes manually, as you are already doing.

Reply:
Intention is to use this layer in future for Contours.